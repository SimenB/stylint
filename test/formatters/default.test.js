'use strict';

const defaultFormatter = require('../../src/formatters/default');
const formatterMockUtils = require('../../src/utils/formatterMockUtils');

const MOCK_FILE_NAME = 'some file.styl';
const MOCK_RULE_ID = 'duplicates';

describe('defaultFormatter', () => {
  let returnValue;
  const report = formatterMockUtils.generateReport([formatterMockUtils.generateWarning(MOCK_FILE_NAME, MOCK_RULE_ID)]);
  const cleanReport = formatterMockUtils.generateReport();

  describe('when the report has no results', () => {
    beforeEach(() => {
      returnValue = defaultFormatter(cleanReport);
    });

    it('should return an empty string', () => {
      expect(returnValue).toBe('');
    });
  });

  describe('when kill is passed in', () => {
    beforeEach(() => {
      returnValue = defaultFormatter(report, {}, true);
    });

    it('should append the report with a warning message', () => {
      expect(returnValue).toMatchSnapshot();
    });
  });

  describe('when max errors or max warnings is 0 or greater', () => {
    beforeEach(() => {
      returnValue = defaultFormatter(report, { maxErrors: 5, maxWarnings: 2 });
    });

    it('should include the max warning and error counts in the report', () => {
      expect(returnValue).toMatchSnapshot();
    });
  });

  describe('when max error or max warning is less than 0', () => {
    beforeEach(() => {
      returnValue = defaultFormatter(report, { maxErrors: -1, maxWarnings: -1 });
    });

    it('should not include the max warning and error counts in the report', () => {
      expect(returnValue).toMatchSnapshot();
    });
  });

  describe('when a message does not include a column', () => {
    beforeEach(() => {
      const noColumnReport = formatterMockUtils.generateReport([
        formatterMockUtils.generateWarning(MOCK_FILE_NAME, MOCK_RULE_ID),
      ]);
      // TODO - This is better encapsulated if it is a part of generateReport.
      noColumnReport.results[0].messages[0].column = null;

      returnValue = defaultFormatter(noColumnReport, {});
    });

    it('should only display the line number for each message', () => {
      expect(returnValue).toMatchSnapshot();
    });
  });
});
