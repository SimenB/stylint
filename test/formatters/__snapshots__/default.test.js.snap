// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaultFormatter when a message does not include a column should only display the line number for each message 1`] = `
"warning: This is not OK
File: some file.styl
Line: 1

Stylint: 0 Errors.
Stylint: 1 Warnings."
`;

exports[`defaultFormatter when kill is passed in should append the report with a warning message 1`] = `
"warning: This is not OK
File: some file.styl
Line: 1:-1

Stylint: 0 Errors.
Stylint: 1 Warnings.
Stylint: Over Error or Warning Limit."
`;

exports[`defaultFormatter when max error or max warning is less than 0 should not include the max warning and error counts in the report 1`] = `
"warning: This is not OK
File: some file.styl
Line: 1:-1

Stylint: 0 Errors.
Stylint: 1 Warnings."
`;

exports[`defaultFormatter when max errors or max warnings is 0 or greater should include the max warning and error counts in the report 1`] = `
"warning: This is not OK
File: some file.styl
Line: 1:-1

Stylint: 0 Errors. (Max Errors: 5)
Stylint: 1 Warnings. (Max Warnings: 2)"
`;
